SuperBuild/External_python-numpy.cmake

```diff
@@ -34,16 +34,16 @@ if(NOT Slicer_USE_SYSTEM_${proj})
   # [/nose]
   # [numpy]
   # Hashes correspond to the following packages:
-  #  - numpy-1.22.1-cp39-cp39-macosx_10_9_x86_64.whl
-  #  - numpy-1.22.1-cp39-cp39-macosx_11_0_arm64.whl
-  #  - numpy-1.22.1-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl
-  #  - numpy-1.22.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
-  #  - numpy-1.22.1-cp39-cp39-win_amd64.whl
-  numpy==1.22.1 --hash=sha256:bcd19dab43b852b03868796f533b5f5561e6c0e3048415e675bec8d2e9d286c1 \
-                --hash=sha256:78bfbdf809fc236490e7e65715bbd98377b122f329457fffde206299e163e7f3 \
-                --hash=sha256:c51124df17f012c3b757380782ae46eee85213a3215e51477e559739f57d9bf6 \
-                --hash=sha256:88d54b7b516f0ca38a69590557814de2dd638d7d4ed04864826acaac5ebb8f01 \
-                --hash=sha256:4ac4d7c9f8ea2a79d721ebfcce81705fc3cd61a10b731354f1049eb8c99521e8
+  #  - numpy-1.22.2-cp39-cp39-macosx_10_14_x86_64.whl
```

Comment:
@jcfr I didn’t notice this jump from macosx_10_9_x86_64 to macosx_10_14_x86_64 in this patch release change of `numpy`. This is https://github.com/numpy/numpy/issues/21066 and why our build fails for our factory-south-macos machine running macOS 10.13 (https://slicer.cdash.org/viewBuildError.php?buildid=2609575), but not on computron running macOS 10.15.

Reply:
While I work on 

> As hot fix, I suggest we pin to `1.22.1`

I stopped the build on both `computron` and `factory-south-macos`. 

I will delete the CDash entries, udpate the local checkout and trigger the script disabling  update of the source tree.

For future reference, here are the error messages:

### CDash report

```
Site:        factory-south-macos.kitware

Build Name:  macOS-clang-10.0.0-64bits-Qt5.15.2-Release
```

```
Error while building " /Volumes/D/P/S-0-build/python-numpy-prefix/src/python-numpy-stamp/python-numpy-install" in target python-numpy 

CMake Error at /Volumes[S-0-build/python-numpy-prefix/src/python-numpy-stamp/python-numpy-install-Release.cmake:49](https://github.com/Slicer/Slicer/blob/master/S-0-build/python-numpy-prefix/src/python-numpy-stamp/python-numpy-install-Release.cmake#L49) (message):
  Command failed: 1

   '/Volumes/D/P/S-0-build/python-install/bin/PythonSlicer' '-m' 'pip' 'install' '--require-hashes' '-r' '/Volumes/D/P/S-0-build/python-numpy-requirements.txt'

  See also

    /Volumes/D/P/S-0-build/python-numpy-prefix/src/python-numpy-stamp/python-numpy-install-*.log
```

### Content of `python-numpy-install-err.log`

```
python-numpy-prefix/src/python-numpy-stamp/python-numpy-install-err.log
```

```
ERROR: THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.
    numpy==1.22.2 from https://files.pythonhosted.org/packages/e9/6c/c0a8130fe198f27bab92f1b28631e0cc2572295f6b7a31e87efe7448aa1c/numpy-1.22.2.zip#sha256=076aee5a3763d41da6bef9565fdf3cb987606f567cd8b104aded2b38b7b47abf (from -r /Volumes/D/P/S-0-build/python-numpy-requirements.txt (line 11)):
        Expected sha256 15efb7b93806d438e3bc590ca8ef2f953b0ce4f86f337ef4559d31ec6cf9d7dd
        Expected     or 59153979d60f5bfe9e4c00e401e24dfe0469ef8da6d68247439d3278f30a180f
        Expected     or 94dd11d9f13ea1be17bac39c1942f527cbf7065f94953cf62dfe805653da2f8f
        Expected     or badca914580eb46385e7f7e4e426fea6de0a37b9e06bec252e481ae7ec287082
        Expected     or d76a26c5118c4d96e264acc9e3242d72e1a2b92e739807b3b69d8d47684b6677
             Got        076aee5a3763d41da6bef9565fdf3cb987606f567cd8b104aded2b38b7b47abf
```

Reply:
> As hot fix, I suggest we pin to `1.22.1`

For future reference, this should be fixed in https://github.com/Slicer/Slicer/pull/6235/commits/94dccb7e3f8583ce2e3e968b66a033d7ea67db61

Reply:
I already have a commit updating deployment target to 10.14 in https://github.com/Slicer/Slicer/pull/6108. I posted there some global usage stats. If you’d like I could break up that PR for a new PR that updates deployment target and restores `numpy` 1.22.2. Or we can consider that existing PR straight up and make the jump to C++17 as requirements needed for upcoming Qt6 support in Slicer.

Reply:
To follow-up on this thread I have updated #6108 to be just the supported platform stuff that drops some platforms that are EOL and moved the bump to C++17 into its own separate PR.

Reply:
https://github.com/numpy/numpy/releases/tag/v1.22.3 has made it clear in the release notes that macOS wheels are supported for 10.14 and later.